src/js/components/heading/Heading.js

```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react';
+import { compose } from 'recompose';
+
+import StyledHeading from './StyledHeading';
+
+import { withTheme } from '../hocs';
+
+import doc from './doc';
+
+const styledComponents = {
+  div: StyledHeading,
+}; // tag -> styled component
+
+class Heading extends Component {
+  static defaultProps = {
+    level: 1,
+  };
+
+  render() {
+    const {
+      level,
+      ...rest
+    } = this.props;
+
+    const tag = `h${level}`;
+    let StyledComponent = styledComponents[tag];
+    if (!StyledComponent) {
+      StyledComponent = StyledHeading.withComponent(tag);
```

Comment:
Callers can only change the level to something from 1-4. So, the tag will be "h1", "h2", "h3", or "h4" only.